client/src/screens/Product.js

```diff
@@ -1,14 +1,24 @@
-import React, { useEffect } from 'react'
+import React, { useEffect, useState } from 'react'
 import { useSelector, useDispatch } from 'react-redux'
 import { useRoute } from '@react-navigation/core'
-import { View, FlatList, Image, Text, Button, Divider } from 'native-base'
+import { View, FlatList, Image, Text, Button, Divider, Input } from 'native-base'
 import Loading from '../components/Loading'
 import { fetchProductById } from '../features/product'
+import { putAnswer } from '../utils/api'
 
 export default ({ navigation }) => {
   const route = useRoute()
-  const { product, loading } = useSelector((state) => state.product)
   const dispatch = useDispatch()
+  const [qestionIdx, setQnIdx] = useState()
+  const [answer, setAnswer] = useState()
+  const { product, loading } = useSelector((state) => state.product)
+
+  const handleAnswerSubmit = async () => {
+    const params = { docId: product._id, answer: answer, questionIdx: qestionIdx }
```

Comment:
```suggestion
    const params = { id: product._id, answer, questionInex }
```

Reply:
This is not fixed yet. `doc` is too general and doesn't add any information to it, so use `id` or `productId`.